src/components/input/input.spec.ts

```diff
@@ -0,0 +1,180 @@
+
+import {
+  it,
+  expect,
+  beforeEach,
+} from '../../core/facade/testing';
+
+import {Component} from 'angular2/core';
+import {By} from 'angular2/platform/browser';
+
+import {
+  MdInput, MdInputDuplicatedHintException, MD_INPUT_DIRECTIVES,
+  MdInputPlaceholderConflictException, MdInputUnsupportedTypeException
+} from './input';
+
+
+export function main() {
+  describe('MdInput', function () {
+    var builder: TestComponentBuilder;
+
+    beforeEach(inject([TestComponentBuilder], function (tcb: TestComponentBuilder) {
+      builder = tcb;
+    }));
+
+    it('creates a native <input> element', injectAsync([], () => {
+      return builder.createAsync(MdInputBaseTestController)
+        .then((fixture) => {
+          fixture.detectChanges();
+          expect(fixture.debugElement.query(By.css('input'))).not.toBeNull();
+        });
+    }));
+
+    it('implements ControlValueAccessor', injectAsync([], () => {
+      return builder.createAsync(MdInputBaseTestController)
+        .then((fixture) => {
+          fixture.detectChanges();
+          fakeAsync(() => {
+            let instance = fixture.componentInstance;
+            let component = fixture.debugElement.query(By.directive(MdInput)).componentInstance;
+            let el: HTMLInputElement = fixture.debugElement.query(By.css('input')).nativeElement;
+
+            instance.model = 'hello';
+            fixture.detectChanges();
+            tick();
+            expect(el.value).toEqual('hello');
+
+            component.value = 'world';
+            fixture.detectChanges();
+            tick();
+            expect(el.value).toEqual('world');
+          })();
+        });
+    }));
+
+    it('counts characters', injectAsync([], () => {
+      return builder.createAsync(MdInputBaseTestController)
+        .then((fixture) => {
+          let instance = fixture.componentInstance;
+          fixture.detectChanges();
+          let inputInstance = fixture.debugElement.query(By.directive(MdInput)).componentInstance;
+          expect(inputInstance.characterCount).toEqual(0);
+
+          instance.model = 'hello';
+          fixture.detectChanges();
+          expect(inputInstance.characterCount).toEqual(5);
+        });
+    }));
+
+    it('copies aria attributes to the inner input', injectAsync([], () => {
+      return builder.createAsync(MdInputAriaTestController)
+        .then((fixture) => {
+          let instance = fixture.componentInstance;
+          fixture.detectChanges();
+          let el: HTMLInputElement = fixture.debugElement.query(By.css('input')).nativeElement;
+          expect(el.getAttribute('aria-label')).toEqual('label');
+          instance.ariaLabel = 'label 2';
+          fixture.detectChanges();
+          expect(el.getAttribute('aria-label')).toEqual('label 2');
+
+          expect(el.getAttribute('aria-disabled')).not.toBeNull();
+        });
+    }));
+
+    it('validates there\'s only one hint label per side', injectAsync([], () => {
```

Comment:
I thought the linting was smart enough to know if you don't need backtick because you don't use `${}`. And double quotes are forbidden by linting too. Will check tomorrow.